File: examples/draw2/src/draw2.c
#include <land/land.h>

typedef struct Game Game;
struct Game
{
    float t;
};

Game *game;

static void game_init(LandRunner *self)
{
    game = calloc(1, sizeof *game);
}

static void game_tick(LandRunner *self)
{
    if (land_key(KEY_ESC))
        land_quit();
    game->t += 1.0 / land_get_frequency();
}

static void game_draw(LandRunner *self)
{
    land_clear(0, 0, 0, 1);

    land_color(0, 0, 1, 1);
    float x = land_display_width() / 2;
    float y = land_display_height() / 2;
    float r = 100 * fabs(sin(game->t));
    land_filled_circle(x - r, y - r, x + r, y + r);
}

land_begin()
{
    land_init();
    land_set_display_parameters(640, 480, 32, 120, LAND_WINDOWED | LAND_OPENGL);
    LandRunner *game_runner = land_runner_new("game",
        game_init, NULL, game_tick, game_draw, NULL, NULL);
    land_runner_register(game_runner);
    land_set_initial_runner(game_runner);
    land_main();
}

File: examples/draw2/src/draw2.py
import global land/land

class Game:
    float t

Game *game

def _init:
    land_alloc(game)

def _done:
    land_free(game)

def _tick:
    if land_key(LandKeyEscape):
        land_quit()
    game->t += 1.0 / land_get_fps()

def _draw:
    land_clear(0, 0, 0, 1)

    float w = land_display_width()
    float h = land_display_height()
    float x = w / 2
    float y = h / 2
    float r = 100 * fabs(sin(game->t))

    land_color(0, 0.25, 0, 1)
    land_line(x, 0, x, h)
    land_line(0, y, w, y)
    
    land_color(0, 0, 1, 1)
    land_filled_circle(x - r, y - r, x + r, y + r)

    land_color(1, 1, 0, 1)
    land_filled_circle(0, 0, 8, 8)
    land_filled_circle(w - 8, 0, w, 8)
    land_filled_circle(w - 8, h - 8, w, h)
    land_filled_circle(0, h - 8, 8, h)

land_standard_example()
